dbl/src/lib.rs

```diff
@@ -38,37 +39,32 @@ pub trait Dbl {
 
 impl Dbl for GenericArray<u8, U8> {
     fn dbl(self) -> Self {
-        let mut val: u64 = unsafe { mem::transmute_copy(&self) };
-        val = val.to_be();
+        let mut val = u64::from_be_bytes(self.into());
+
         let a = val >> 63;
         val <<= 1;
         val ^= a * C64;
-        unsafe { mem::transmute(val.to_be()) }
+
+        val.to_be_bytes().into()
     }
 
     fn inv_dbl(self) -> Self {
-        let mut val: u64 = unsafe { mem::transmute_copy(&self) };
-        val = val.to_be();
+        let mut val = u64::from_be_bytes(self.into());
 
         let a = val & 1;
         val >>= 1;
         val ^= a * ((1 << 63) ^ (C64 >> 1));
 
-        unsafe { mem::transmute(val.to_be()) }
-    }
-}
-
-#[inline(always)]
-fn to_be(val: &mut [u64]) {
-    for v in val.iter_mut() {
-        *v = v.to_be();
+        val.to_be_bytes().into()
     }
 }
 
 impl Dbl for GenericArray<u8, U16> {
     fn dbl(self) -> Self {
-        let mut val: [u64; 2] = unsafe { mem::transmute_copy(&self) };
-        to_be(&mut val);
+        let mut val = [0u64; 2];
+        for (s, v) in self.chunks_exact(size_of::<u64>()).zip(val.iter_mut()) {
+            *v = u64::from_be_bytes(s.try_into().unwrap());
+        }
```

Comment:
I think we can replace it simply with:
```rust
let mut val = [
    u64::from_be_bytes(self[..8].try_into().unwrap()),
    u64::from_be_bytes(self[8..].try_into().unwrap()),
];
```

Reply:
Updated for the methods `dbl` and `inv_dbl` having a block size of 128 bits. 

For the block size of 256 bits I kept the for loop. IMO its a bit better readable, but can change it as well.